onnxruntime/core/providers/openvino/openvino_graph.cc

```diff
@@ -0,0 +1,369 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#include <iostream>
+#include <cstdlib>
+#include <map>
+#include <string>
+#include <memory>
+#include <cstdlib>
+#include <fstream>
+#include <Python.h>
+
+#include <inference_engine.hpp>
+#include <ie_builders.hpp>
+
+#include "core/graph/graph.h"
+#include "core/framework/tensorprotoutils.h"
+
+#include "openvino_graph.h"
+
+namespace openvino_ep {
+
+OpenVINOGraph::OpenVINOGraph(onnxruntime::Node* fused_node, std::string /*device_info*/) {
+	//TODO: parse device info to obtain the following values
+
+
+  device_id_ = "CPU";
+  precision_ = InferenceEngine::Precision::FP32;
+  std::string precision_str = "FP32";
+
+#ifdef OPENVINO_CONFIG_CPU_FP32
+	device_id_ = "CPU";
+	precision_ = InferenceEngine::Precision::FP32;
+	precision_str = "FP32";
+#endif
+#ifdef OPENVINO_CONFIG_GPU_FP32
+	device_id_ = "GPU";
+	precision_ = InferenceEngine::Precision::FP32;
+	precision_str = "FP32";
+#endif
+#ifdef OPENVINO_CONFIG_GPU_FP16
+	device_id_ = "GPU";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+#ifdef OPENVINO_CONFIG_MYRIAD
+	device_id_ = "MYRIAD";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+#ifdef OPENVINO_CONFIG_VAD_R
+	device_id_ = "HDDL";
+	precision_ = InferenceEngine::Precision::FP16;
+	precision_str = "FP16";
+#endif
+
+  std::cout<< "[OpenVINO-EP] device:" << device_id_ << std::endl;
+  std::cout<< "[OpenVINO-EP] precision:" << precision_str << std::endl;
+
+
+	num_inf_reqs_ = (device_id_ == "HDDL") ? 8 : 1;
+
+	fused_node_ = fused_node;
+
+
+
+    cnn_network_ = BuildCNNNetworkWithMO();
+
+  // TODO: make this a debug option
+  // Uncomment the below code to enable dumping IE CNN graphs.
+  //std::string file_name = "./conv_" + fused_node->Name();
+  //cnn_network_->serialize( file_name+".xml", file_name+".bin");
+
+
+  infer_requests_ = GetExecutableHandle(cnn_network_, device_id_, precision_);
+}
+
+std::vector<std::string> OpenVINOGraph::GetEnvLdLibraryPath() {
+    std::string plugin_path = std::getenv("LD_LIBRARY_PATH");
+    std::vector<std::string> paths;
+    std::string token;
+    std::istringstream tokenStream(plugin_path);
+    char delimiter = ':';
+
+    while (std::getline(tokenStream , token, delimiter)) {
+      paths.push_back(token);
+    }
+    return paths;
+}
+
+std::shared_ptr<InferenceEngine::CNNNetwork> OpenVINOGraph::BuildCNNNetworkWithMO() {
+
+    PyObject *pModule, *pOutput;
+    PyObject *pArg1, *pArg2;
+    PyObject *pFunc = nullptr;
+
+    Py_Initialize();
+
+    pModule = PyImport_ImportModule("openvino_mo");
```

Comment:
is this a temporary solution? i didn't realize we would need to call python modules.
aren't there native api's which can be used?

Reply:
Current design requires invoking the python based Model Optimizer module for graph translation.